database/tx_queue.go

```diff
@@ -0,0 +1,49 @@
+package database
+
+import (
+	"time"
+	"sync"
+	
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	"sort"
+)
+
+const awaitingTxsTimeInMs = 500
+
+type txQueue struct {
+	lastUpdate time.Time
+	txs        []*ethTypes.Transaction
+	mtx        sync.Mutex
+}
+
+func (q *txQueue) addTx(tx *ethTypes.Transaction) {
+	q.mtx.Lock()
+	defer q.mtx.Unlock()
+	q.lastUpdate = time.Now()
+	q.txs = append(q.txs, tx)
+	sort.Slice(q.txs, func(i, j int) bool { 
+		return q.txs[i].Nonce() > q.txs[j].Nonce() 
+	})
```

Comment:
[critical to consider] It seems there are 3 problems with this implementation:

- By design, every TX is delayed by 0.5s
- In case of a busy hour, if there is a new TX incoming to a sentry every <0.5s, we never broadcast and process ANY TX as we keep waiting for a next one

Proposed solution: The queue should be account based. In that case, if someone is producing a lot of TXs, it doesn't block other accounts.

Reply:
Good catch! although I am very inclined to the proposed solution. Let discuss alternatives